### PR TITLE
Fix some crash on exits

### PR DESCRIPTION
**Pull request by sethalves**
- fix some crashes-upon-exit

### CODE REVIEW

**Comment by sethalves** [comment]
to test:
- no functional changes, especially in plugin related areas

**Comment by hifi-gustavo** [comment]
Build console output for commit: 9868ddd6029d69d5846e17bc6497ecd830c59f2e available at: https://gist.github.com/4deaaab41912057b3f32c6e7a5a3f175<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9063-9868ddd.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9063-9868ddd.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 9b4a8427e6cfa9d3d689b01c00926d537f49adae available at: https://gist.github.com/4deaaab41912057b3f32c6e7a5a3f175<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9063-9b4a842.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9063-9b4a842.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: f8fb5b970928487a58b898b1556e61a6e026af60 available at: https://gist.github.com/4deaaab41912057b3f32c6e7a5a3f175<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9063-f8fb5b9.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9063-f8fb5b9.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 45bcf626aadc71092030e8d9c4b9ae35a1e4c9b8 available at: https://gist.github.com/4deaaab41912057b3f32c6e7a5a3f175<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9063-45bcf62.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9063-45bcf62.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 359f05426cc25b353548602fa18e1e55bc6f5026 available at: https://gist.github.com/4deaaab41912057b3f32c6e7a5a3f175<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9063-359f054.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9063-359f054.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 946dc256f092bb2c520882e3485786f63e950175 available at: https://gist.github.com/4deaaab41912057b3f32c6e7a5a3f175<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9063-946dc25.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9063-946dc25.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: da83fd9fa46683996df318cc256a4f269629f21c available at: https://gist.github.com/4deaaab41912057b3f32c6e7a5a3f175<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9063-da83fd9.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9063-da83fd9.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 645134e4b06f2fc4f05dee5a11af3d77ca71ec89 available at: https://gist.github.com/4deaaab41912057b3f32c6e7a5a3f175<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9063-645134e.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9063-645134e.dmg

File: libraries/display-plugins/src/display-plugins/OpenGLDisplayPlugin.cpp [hunk 1]
```diff
@@ -758,7 +758,7 @@ void OpenGLDisplayPlugin::render(std::function<void(gpu::Batch& batch)> f) {
 
 
 OpenGLDisplayPlugin::~OpenGLDisplayPlugin() {
-    qDebug() << "Destroying OpenGLDisplayPlugin";
+    // qDebug() << "Destroying OpenGLDisplayPlugin";
```
Comment (ZappoMan) [inline libraries/display-plugins/src/display-plugins/OpenGLDisplayPlugin.cpp]: Let's just remove this.

File: libraries/display-plugins/src/display-plugins/hmd/DebugHmdDisplayPlugin.h [hunk 2]
```diff
@@ -13,7 +13,7 @@ class DebugHmdDisplayPlugin : public HmdDisplayPlugin {
     using Parent = HmdDisplayPlugin;
 
 public:
-    const QString& getName() const override { return NAME; }
+    const QString getName() const override { return NAME; }
```
Comment (ZappoMan) [inline libraries/display-plugins/src/display-plugins/hmd/DebugHmdDisplayPlugin.h]: no point in the const on these... but not a blocker

**Comment by ZappoMan** [comment]
Can you explain these fixes and these crashes? Do you have a solid explanation? or is this belt and suspenders?

**Comment by sethalves** [comment]
A couple of them are just commented out "destroying this" logging, which appears to attempt to print too late and crashes.  All the QString --> char * stuff I can't be as sure about.  I get crashes, and I can see that it's a QString being destructed.  If I move aside the plugins, I get a clean exit.  If I leave the plugins in place and replace their global/static QStrings with char * I get a clean exit.

It's voodoo.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 92e24e6ded96f9e35e92fb241211fe69982ce935 available at: https://gist.github.com/4deaaab41912057b3f32c6e7a5a3f175<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9063-92e24e6.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9063-92e24e6.dmg

**Comment by OmegaHeron** [comment]
Testing Windows 10
QA Pass

General smoke test good.  Ran this PR a few times, visited multiple places, used audio, toggled HMD/desktop modes and quit -- no crashes.  That said - I've not had any crashes on close for a long time with Windows.  Also checked logs for any spam/oddities - none found.

**Pull request merged**